torch/utils/checkpoint.py

```diff
@@ -94,7 +117,7 @@ def backward(ctx, *args):
                 torch.set_rng_state(ctx.fwd_cpu_state)
                 if ctx.had_cuda_in_fwd:
                     set_device_states(ctx.fwd_gpu_devices, ctx.fwd_gpu_states)
-            detached_inputs = detach_variable(inputs)
+            detached_inputs = detach_variable(tuple(inputs))
```

Comment:
This function is just used here no? Maybe just update the function to take a list as input instead of a tuple?